seed/static/seed/js/controllers/confirm_column_settings_modal_controller.js

```diff
@@ -0,0 +1,326 @@
+      });
+
+      // parse proposed changes to create change summary to be presented to user
+      var all_changed_settings = ["column_name"];  // add column_name to describe each row
+      $scope.change_summary_data = _.reduce(proposed_changes, function (summary, value, key) {
+        var column = _.find($scope.all_columns, {id: parseInt(key)});
+        var change = _.pick(_.cloneDeep(column), ['column_name']);
+
+        // capture changed setting values
+        summary.push(_.merge(change, value));
+
+        // capture corresponding settings columns
+        all_changed_settings = _.concat(all_changed_settings, _.keys(value));
+        return summary;
+      }, []);
+
+      var base_summary_column_defs = [
+        {
+          field: 'column_name'
+        },
+        {
+          field: 'display_name',
+          displayName: 'Display Name Change',
+        },
+        {
+          field: 'data_type',
+          displayName: 'Data Type Change',
+        },
+        {
+          field: 'merge_protection',
+          displayName: 'Merge Protection Change',
+          cellTemplate: '<div class="ui-grid-cell-contents text-center">' +
+            '<input type="checkbox" class="no-click" ng-show="{$ row.entity.merge_protection != undefined $}" ng-checked="{$ row.entity.merge_protection === \'Favor Existing\' $}" style="margin: 0px;">' +
+            '</div>',
+        },
+        {
+          field: 'is_matching_criteria',
+          displayName: 'Matching Criteria Change',
+          cellTemplate: '<div class="ui-grid-cell-contents text-center">' +
+            '<input type="checkbox" class="no-click" ng-hide="{$ row.entity.is_matching_criteria === undefined $}" ng-checked="{$ row.entity.is_matching_criteria === true $}" style="margin: 0px;">' +
+            '</div>',
+        },
+      ]
+
+      var unique_summary_columns = _.uniq(all_changed_settings);
+      $scope.change_summary_column_defs = _.filter(base_summary_column_defs, function (column_def) {
+        return _.includes(unique_summary_columns, column_def.field);
+      });
+
+      $scope.change_summary = {
+        data: $scope.change_summary_data,
+        columnDefs: $scope.change_summary_column_defs,
+        enableColumnResizing: true,
+        minRowsToShow: Math.min($scope.change_summary_data.length, 5),
+      };
+
+      // By default, assume matching criteria isn't being updated to exclude PM Property ID
+      // And since warning wouldn't be shown in that case, set "acknowledged" to true.
+      $scope.checks = {
+        matching_criteria_excludes_pm_property_id: false,
+        warnings_acknowledged: true,
+      }
+
+      // Check if PM Property ID is actually being removed from matching criteria
+      if (_.find($scope.change_summary_data, {column_name: "pm_property_id", is_matching_criteria: false})) {
+        $scope.checks.matching_criteria_excludes_pm_property_id = true;
+        $scope.checks.warnings_acknowledged = false;
+      }
+
+      // Preview
```

Comment:
There are some duplicate lines of code between this and the cycles list controller but there are definitely some necessary and nuanced differences. It's likely possible to DRY this up a bit by abstracting the two but I didn't place a high priority on it as it's maybe not the most cost efficient talk. I'm happy to revisit this later though